openmc/lib/dagmc.py

```diff
@@ -0,0 +1,62 @@
+import sys
+from ctypes import c_int, c_int32, POINTER, c_size_t
+
+import numpy as np
+
+from . import _dll
+from .error import _error_handler
+
+
+# DAGMC functions
+_dll.openmc_get_dagmc_cell_ids.argtypes = [c_int32, POINTER(c_int32), POINTER(c_size_t)]
+_dll.openmc_get_dagmc_cell_ids.restype = c_int
+_dll.openmc_get_dagmc_cell_ids.errcheck = _error_handler
+_dll.openmc_dagmc_universe_get_num_cells.argtypes = [c_int32, POINTER(c_size_t)]
+_dll.openmc_dagmc_universe_get_num_cells.restype = c_int
+_dll.openmc_dagmc_universe_get_num_cells.errcheck = _error_handler
+
+
+def get_dagmc_cell_ids(dagmc_id):
```

Comment:
I know this would be a significant change, and one that I'm open to leaving to a subsequent PR, but I wonder if a `DAGMCUniverse` class implemented here could benefit from a lot of existing functionality of the `Universe` class.

Reply:
Fantastic! That will be rather nifty I think.